### PR TITLE
AtSpi2: Delete AtSpiElement class.

### PR DESCRIPTION
**Pull request by madewokherd**
This does add the `child_count` variable, since we can no longer use the `AtSpiElement` type to determine if a child is managed by `AccessibleProvider`. I have something that removes it in favor of another system in my feature branch here: https://github.com/madewokherd/xalia/commit/c5026af967135da7a2171827bfe2a77ec29c721c

### CODE REVIEW

File: xalia/AtSpi2/ActionProvider.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        }
+
+        public void TrackedPropertyChanged(UiDomElement element, string name, UiDomValue new_value)
+        {
+        }
+
+        public bool UnwatchProperty(UiDomElement element, GudlExpression expression)
+        {
+            return false;
+        }
```
[the remark this answers is not in the document]
Comment (madewokherd) [reply xalia/AtSpi2/ActionProvider.cs:83]: Maybe. I was a bit too annoyed at the time with the lack of multiple inheritance to consider that maybe a base class would've made sense.

File: xalia/AtSpi2/ActionProvider.cs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        {
+            if (expression is IdentifierExpression id)
+            {
+                switch (id.Name)
+                {
+                    case "spi_action":
+                        if (!fetching_actions)
+                        {
+                            fetching_actions = true;
+                            Utils.RunTask(FetchActions());
```
[the remark this answers is not in the document]
Comment (madewokherd) [reply xalia/AtSpi2/ActionProvider.cs:95]: This is intended. It's a pattern I use for properties that aren't expected to change throughout the object's lifetime, so once a fetch has been started there's no reason to do it again even if it fails.

[the remark this answers is not in the document]
Comment (Conmanx360) [reply xalia/AtSpi2/ActionProvider.cs:95]: Nitpicky, but the variable name makes it seem like something that could start/stop, might potentially be clearer if it was `fetched_actions` or `actions_fetched`, but that's not really a blocker.

File: xalia/AtSpi2/ActionProvider.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Conmanx360) [reply xalia/AtSpi2/ActionProvider.cs:83]: Fair enough. :)

**Comment by Conmanx360** [comment]
Other than my nitpicks, I have no major complaints. :)

File: xalia/AtSpi2/ActionProvider.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (madewokherd) [reply xalia/AtSpi2/ActionProvider.cs:95]: But they might not have been fetched yet. It's really "started fetching actions", but that's unwieldy.

File: xalia/AtSpi2/ActionProvider.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (madewokherd) [reply xalia/AtSpi2/ActionProvider.cs:83]: I'll put this on my todo list as a later enhancement. I should at least get it done before adding any more provider classes.

File: xalia/AtSpi2/ActionProvider.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Conmanx360) [reply xalia/AtSpi2/ActionProvider.cs:95]: Given I don't really have any good suggestions on a better name, it's probably good to just ignore this. I agree other alternatives are unwieldy, the only ones I can really come up with might be `fetching_actions_task_started`, but at that point the variable name is relatively long and not exactly pretty.

**Pull request merged**